### ISSUE TITLE
[a11y]: Tag [tag content] - list should be proved to tag content

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#15267]
### Package

@carbon/react

### Browser

FF, Safari & iPhone

### Package version

@carbon/react@1.41.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

VO & JAWS

### Package

@carbon/react

### Browser

FF, Safari & iPhone

### Package version

@carbon/react@1.41.1

### Automated testing tool and ruleset

Manual QA

### Assistive technology

VO & JAWS

### Description
List implementation should be provided to the tag content

There is no list implemented for the tags content. Screen reader is just reading the tags.

### Recommended Fix
List should be defined properly.

People who uses screen reader will not be able to understand the structure of the content in the page.

### WCAG 2.1 Violation
1.3.1 Info and Relationships

**Comment by kennylam** [comment]
Evaluate the Storybook example vs the native output of the component to determine if the bug is at the component level or if the story is authored incorrectly - https://www.w3.org/WAI/WCAG22/Understanding/info-and-relationships.
